app/components/NFTDetails.tsx

```diff
@@ -1,4 +1,4 @@
-"use server";
+"use client";
```

Comment:
Why did you change this to a client component? For speed reasons we want to as much as possible on the server side.